apps/bfd-shared-utils/src/main/java/gov/cms/bfd/sharedutils/SecurityLabels.java

```
@@ -0,0 +1,89 @@
+package gov.cms.bfd.sharedutils;
+
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.yaml.snakeyaml.Yaml;
+
+/**
+ * Utility class for reading and organizing security label data from YAML files.
+ *
+ * <p>This class provides methods to parse YAML files containing structured metadata entries (such
+ * as system, code, startDate, and endDate), and group them by the "system" key. It is primarily
+ * used to support security labeling and classification tasks in applications that rely on
+ * standardized metadata formats.
+ *
+ * <p>Example YAML structure:
+ *
+ * <pre>
+ * - system: http://hl7.org/fhir/sid/icd-10-cm
+ *   code: F19.159
+ *   startDate: 1970-01-01
+ *   endDate: ACTIVE
+ * </pre>
+ *
+ * @version 1.0
+ */
+public class SecurityLabels {
+
+  /**
+   * Reads a YAML file containing entries with keys such as "system", "code", "startDate", and
+   * "endDate", and groups the entries by the "system" key.
+   *
+   * @param inputStream the path to the YAML file to be read
+   * @return a map where each key is a "system" value, and the value is a list of entries (maps)
+   *     associated with that system
+   */
+  public static Map<String, List<Map<String, Object>>> readYamlToMap(InputStream inputStream) {
+    Yaml yaml = new Yaml();
+    Map<String, List<Map<String, Object>>> resultMap = new HashMap<>();
+
+    try {
+      List<Map<String, Object>> yamlData = yaml.load(inputStream);
+
+      for (Map<String, Object> entry : yamlData) {
+        String system = (String) entry.get("system");
+        resultMap.computeIfAbsent(system, k -> new ArrayList<>()).add(entry);
+      }
+    } catch (Exception e) {
+      e.printStackTrace();
```

This should use a proper logger and it should probably re-throw the error. We want the application to fail to start if this doesn't work since this is critical functionality.